### PR TITLE
Always return a valid semver from fu_common_version_ensure_semver()

### PR DESCRIPTION
**Pull request by hughsie**
Return NULL if not valid, and handle tilde and dash where required.

Type of pull request:
- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [X] Code fix
- [ ] Feature
- [ ] Documentation

### CODE REVIEW

File: libfwupdplugin/fu-self-test.c [hunk 1]
```diff
@@ -1287,6 +1287,34 @@ fu_common_strstrip_func (void)
...
+		{ "1..2",		"1.2" },
+		{ "CBET1.2.3",		"1.2.3" },
+		{ "1.2.3alpha",		"1.2.3" },
+		{ "5",			"5" },
+		{ "\t5\n",		"5" },
+		{ "coreboot-unknown",	NULL },
+		{ "",			NULL },
+		{ " ",			NULL },
+		{ NULL,			NULL }
+	};
```
Comment (hughsie) [inline libfwupdplugin/fu-self-test.c:1311]: @superm1 any more edge cases you can think of?

Comment (superm1) [reply libfwupdplugin/fu-self-test.c:1311]: Hex?

**Review by superm1 — COMMENTED** [review]
Maybe ending in a -?

File: libfwupdplugin/fu-self-test.c (hunk 1 above)
Comment (hughsie) [reply libfwupdplugin/fu-self-test.c:1336]: Good idea, thanks.
